### DISCUSSION TITLE
table with links in first column to pages that will be generated later

### DISCUSSION THREAD

**Discussion by tvanees-ohpen-com** [py-pdf/fpdf2#821] [Q&A]
I have to document a hierarchical structure in a pdf file . At the highest level I would want to make a table with entries (containing links) to pages describing objects on the second level and on each second level page, I would want to make a table with entries (containing links) to pages describing objects on the third level. The tables on each level might span multiple pages, so without additional logic, it is difficult to predict on which page the description of an object at a certain level starts.

My first question is, how to I convert a text in a column in a table to a link and my second question is how can I make links to pages that still need to be generated.

One solution is to build everything in internal structures, calculate the page numbers etc and then from that internal structure, write the pdf. However that feels a bit complex.

**Comment by Lucas-C** [comment]
> My first question is, how to I convert a text in a column in a table to a link

Several methods allow you to create links:
https://pyfpdf.github.io/fpdf2/Links.html

> how can I make links to pages that still need to be generated

You can use [FPDF.add_link()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.add_link) for that.
Check out the unit test `test_later_call_to_set_link()` in https://github.com/PyFPDF/fpdf2/blob/master/test/test_links.py
for an example on how to do so.

> One solution is to build everything in internal structures, calculate the page numbers etc and then from that internal structure, write the pdf. However that feels a bit complex.

This is a solution for sure, but with `FPDF.add_link()` you shouldn't have to do that.

**Reply by tvanees-ohpen-com** [reply]
Thanks, answer regarding links to pages that still need to be generated is very useful.
With respect to the first question, the 'hyperlink' needs to be inside a cell in a table. while pdf.cell has a link attribute, row.cell seems not to have a link attribute.

**Reply by Lucas-C** [reply]
I opened PR https://github.com/PyFPDF/fpdf2/pull/825 to implement this

**Reply by Lucas-C** [reply]
It is now merged, you can test it before the next release using this command:
`pip install git+https://github.com/PyFPDF/fpdf2.git@master`
